sql/database.go

```diff
@@ -51,3 +53,39 @@ func (p *planner) getDatabaseDesc(name string) (*DatabaseDescriptor, error) {
 	}
 	return &desc, nil
 }
+
+// getCachedDatabaseDesc looks up the database descriptor given its name in the
+// descriptor cache.
+func (p *planner) getCachedDatabaseDesc(name string) (*DatabaseDescriptor, error) {
+	if name == SystemDB.Name {
+		return &SystemDB, nil
+	}
+
+	nameKey := databaseKey{name}
+	var nameVal *roachpb.Value
+	if p.systemConfig != nil {
+		nameVal = p.systemConfig.GetValue(nameKey.Key())
+	}
```

Comment:
if the systemConfig can be nil, better to error out here directly to get a more descriptive error.